src/mscorlib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -643,18 +685,90 @@ internal String ToString(TraceFormat traceFormat)
                         }
                     }
 
-                    if (sf.GetIsLastFrameFromForeignExceptionStackTrace())
+                    if (!FormattingOptions.HasFlag(StackTraceFormattingOptions.ExcludeDispatchBoundaries) && sf.GetIsLastFrameFromForeignExceptionStackTrace())
                     {
-                        sb.Append(Environment.NewLine);
+                        sb.AppendLine();
                         sb.Append(SR.Exception_EndStackTraceFromPreviousThrow);
                     }
                 }
             }
 
             if (traceFormat == TraceFormat.TrailingNewLine)
-                sb.Append(Environment.NewLine);
+            {
+                sb.AppendLine();
+            }
+        }
 
-            return sb.ToString();
+        private static Type ResolveIteratorMethod(Type t, ref MethodBase mb)
+        {
+            Type dt = t.DeclaringType;
+            if (dt != null)
+            {
+                string fullName = t.FullName;
+                var start = fullName.LastIndexOf('<');
```

Comment:
Note this assumes a C#-specific mangled name format. I've suggested in the past that the compiler should emit an attribute on the state machine that resolves the method (e.g. by including a signature string or at least a name).

Also, I think this is a non-negligible computation for a "nice to have" feature (getting the list of argument types): resolving multiple `MemberInfo`s, allocating an array, allocating attributes...

Reply:
> Note this assumes a C#-specific mangled name format. 

Need to check the names used across the languges, worse case it doesn't look it up and leaves the name alone

> The compiler should emit an attribute on the state machine that resolves the method

Would be useful

> Also, I think this is a non-negligible computation for a "nice to have" feature

Its an opt-in; but without it and without line numbers it makes it hard to workout which of the overloads in the example program is being called if the name alone was cleaned

```csharp
async Task<int> TaskTAsync(long count);
async Task<int> TaskTAsync(int count);
async Task<int> TaskTAsync(long count, string name);
```

Reply:
VB is something like
```vb
VB$StateMachine_3_BarAsync.MoveNext()
```

Reply:
Will do some throughput and allocation testing, but 
```
Program.TaskTAsync(Int32 count)
```
Adds a lot of value over
```
Program.<TaskTAsync>d__11.MoveNext()
```
as you know which overload it was rather than having to manual track back via the il and try to eyeball it

Reply:
Full prettification has an insignificant cost compared to the exception handling

Reply:
> compiler should emit an attribute on the state machine that resolves the method

Is there a roslyn issue for this? Should I raise one?

Reply:
Please do not parse compiler generated names.

Reply:
I'd recommend leaving the type and method name pretty-printing for a separate PR as it needs more design and discussion.

Reply:
Agree with @tmat here. I'm behind the goal, but lets get the incremental progress on StackTraceHidden in the bag first. I imagine handling the compiler generated state machines in a robust way that doesn't tie the compiler's hands will take a little longer, using coordinated changes in both compiler and BCL.